src/common/Widget.js

```diff
@@ -452,7 +452,13 @@
     Widget.prototype.rowToObj = function (row) {
         var retVal = {};
         this.columns().forEach(function(col, idx) {
-            retVal[col] = row[idx];
+            var item;
+            if (row[idx] instanceof Widget) {
+                item = row[idx]._class;
```

Comment:
@GordonSmith I was just about to ask you about this...  I did this for testing purposes, because in the click event JSON.stringify returns a circular reference when a widget is included.  Obviously this needs to return the widget itself, so the question is really how should I handle the logging of the click event?